bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/internal/CustomIconProvider.java

```diff
@@ -45,19 +53,19 @@ public InputStream getIcon(String iconName) {
         return null;
     }
 
-    private File getIconFile(String iconName) {
-        File folder = new File(ConfigConstants.getConfigFolder() + File.separator + "icons");
-        File file = new File(folder, iconName + ".png");
-        if (file.exists()) {
-            return file;
-        } else {
-            folder = new File("runtime" + File.separator + "icons");
-            file = new File(folder, iconName + ".png");
+    @Override
+    protected boolean hasResource(String iconSetId, String resourceName) {
+        File file = getIconFile(resourceName, iconSetId);
+        return file != null;
+    }
 
-            if (file.exists()) {
-                return file;
-            }
-        }
-        return null;
+    @Override
+    public Set<IconSet> getIconSets(Locale locale) {
+        return Collections.emptySet();
```

Comment:
You would not have any meta-info from the folders and the custom provider is only meant to augment existing icon sets by custom icons.